### PR TITLE
Exploding messages: timer to the right

### PR DESCRIPTION
**Pull request by buoyad**
The styling is still a little rough around the message container & send indicator. It'll probably be cleaned a little in a later PR, when the `EXPLODED` annotation is added. This is the timer counting down and the boom icon as things explode:

image

 r? @keybase/react-hackers

### CODE REVIEW

File: shared/chat/conversation/messages/wrapper/exploding-meta.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    return diff - Math.floor(diff / oneHourInMs) * oneHourInMs
+  }
+  if (diff > oneMinuteInMs) {
+    return diff - Math.floor(diff / oneMinuteInMs) * oneMinuteInMs
+  }
+  // less than a minute, check every second
+  return 1000
+}
+
+const formatTimeDifference = (d: number): string => {
```
Comment (adamjspooner) [inline shared/chat/conversation/messages/wrapper/exploding-meta.js]: Can we get this into a global helper of some kind?

Comment (chrisnojima) [reply shared/chat/conversation/messages/wrapper/exploding-meta.js]: use util/timestamp, it might exist or it should go in there

**Comment by adamjspooner** [comment]
What are the performance implications of this?

**Comment by buoyad** [comment]
@adamjspooner the timer updates as infrequently as possible (due to `getLoopInterval`), so the most performance heavy case is a very active chat with a relatively short explode time, where you can have a lot of these timers going off every second. I tested that case and didn't notice a difference in performance.

**Pull request merged**